fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -443,11 +449,18 @@ namespace fileBackup {
 		RangeFileWriter(Reference<IBackupFile> file = Reference<IBackupFile>(), int blockSize = 0) : file(file), blockSize(blockSize), blockEnd(0), fileVersion(1001) {}
 
 		// Handles the first block and internal blocks.  Ends current block if needed.
-		ACTOR static Future<Void> newBlock(RangeFileWriter *self, int bytesNeeded) {
+		// The final flag is used in simulation to pad the file's final block to a whole block size
+		ACTOR static Future<Void> newBlock(RangeFileWriter *self, int bytesNeeded, bool final = false) {
 			// Write padding to finish current block if needed
 			int bytesLeft = self->blockEnd - self->file->size();
 			if(bytesLeft > 0) {
-				Void _ = wait(self->file->append((uint8_t *)paddingFFs.data(), bytesLeft));
+				state Value paddingFFs = makePadding(bytesLeft);
+				Void _ = wait(self->file->append(paddingFFs.begin(), bytesLeft));
```

Comment:
I think we do not have to use "Void _ " any more in the master branch. 
Is this a must for the old branch you worked on?